### PR TITLE
Add python-isal package

### PR DESCRIPTION
**Pull request by rhpvorderman**
This package enables python bindings for the isa-l library. This library provides a significant speedup (2x+) when compressing and decompressing gzipped files.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rhpvorderman** [comment]
@conda-forge/help-python-c 
I cannot get this to compile because libc.so.6 cannot be found. What is the best solution? I read that libc is not provided by conda-forge. I guess I can remove any of the libc references in the code for version 0.2.0 if that is required.

File: recipes/python-isal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  # isa-l package only available on linux as of yet.
+  skip: True  # [not linux]  
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - isa-l >=2.26
+    - cython
+    - {{ compiler('c') }}
```
Comment (xhochy) [inline recipes/python-isal/meta.yaml:25]: This needs to be in `build:`, not `host`

Comment (rhpvorderman) [reply recipes/python-isal/meta.yaml:25]: Thanks. Unfortunately this does not solve the problem with the libc.so and libpthread.so linking.

Comment (rhpvorderman) [reply recipes/python-isal/meta.yaml]: I was able to fix it. Apparently only the c compiler needs to be in build, the rest needs to be in build. This is how similar packages (also build with cython) work. Thanks for the hint.

**Comment by rhpvorderman** [comment]
@conda-forge/help-python-c This package is ready for merging.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**